.github/workflows/pull-request.yml

```diff
@@ -0,0 +1,58 @@
+name: pull-request
+on:
+  workflow_dispatch:
+  pull_request:
+    paths:
+    - source/**
+    - clik/**
+    - modules/**
+    - examples/**
+    - cmake/**
+    - hal/**
+    - .github/actions/do_build_ock/**
+    - .github/actions/setup_ubuntu_build/**
+    - .github/workflows/run_pr_tests.yml
+    - .github/workflows/pull-request.yml
+    - CMakeLists.txt
+concurrency:
+  group: ${{ github.workflow }}-${{ github.event.pull_request.number || github.ref }}
+  cancel-in-progress: true
+jobs:
+
+############### JOB   mr-ubuntu-gcc-x86_64-riscv-fp16-cl3-0-unitcl_vecz:
+
+  mr-ubuntu-gcc-x86_64-riscv-fp16-cl3-0-unitcl_vecz:
+    runs-on: ubuntu-22.04
+    timeout-minutes: 60
+    steps:
+    - name: Checkout repo
+      uses: actions/checkout@v4.1.0
+      with:
+        path: code
+    - run: cp -R code/.github .github
+    - name: setup-ubuntu
+      uses: ./.github/actions/setup_ubuntu_build
+      with:
+        llvm_version: '18'
+        llvm_build_type: RelAssert
+    - run: echo WORKSPACE is $GITHUB_WORKSPACE && echo PWD is `pwd` && ls -al
+    - name: build ock
+      uses: ./.github/actions/do_build_ock
+      with:
+        path: $GITHUB_WORKSPACE/code
+        install_dir: $GITHUB_WORKSPACE/install
+        build_targets: check-ock-UnitCL-group-vecz
+        mux_targets_enable: riscv
+        external_compiler_dirs: ${{ github.workspace }}/code/examples/refsi/refsi_m1/compiler/refsi_m1
```

Comment:
This maybe allows us to remove the riscv job we already have since I think that is also m1.

Reply:
fyi ... the single difference between the 2 jobs is
new PR: `-DCA_MUX_COMPILERS_TO_ENABLE=riscv`
existing: `-DCA_MUX_COMPILERS_TO_ENABLE=refsi_m1`

Reply:
hmm, that is odd. The old one seems more correct to me.

Reply:
This becomes a little knotted in that case. We've actually got 2 related Gitlab jobs
```
mr-ubuntu-gcc-x86_64-riscv-fp16-cl3-0-unitcl_vecz
mr-ubuntu-gcc-x86_64-riscv-fp16-cl3-0
```
... both with provisional migrations to Github. (first one under this PR). And we've got the existing Github riscv job mentioned above.
I don't have a clear view on which, if any, are correct or how to resolve them otherwise.
Any suggestions  - with a view to unblocking this PR specifically?

Reply:
I think if it says g1 it shouldn't point to m1 compiler, so just drop that.
We can look at the others as they come in.

Reply:
m1 compiler dropped